cli/lib/gateway.js

```diff
@@ -1,104 +1,127 @@
+  const consoleLogPath = configLocations.getConsoleLogPath(options.org, options.env);
+  const errorLogPath = configLocations.getErrorLogPath(options.org, options.env);
+
   const keys = { key: options.key, secret: options.secret };
   const args = { target: cache, keys: keys,pluginDir:options.pluginDir };
-  const that = this;
-
-  if (cluster.isMaster) {
-    edgeconfig.get({ source: source, keys: keys }, function (err, config) {
-      
-      if(options.port){
-        config.edgemicro.port = parseInt(options.port);
-      }
 
-      if (err) {
-        const exists = fs.existsSync(cache);
-        console.error("failed to retieve config from gateway. continuing, will try cached copy..");
-        console.error(err);
-        if(!exists){
-          return cb('cache configuration '+cache+' does not exist. exiting.');
-        }else{
-          console.log('using cached configuration from %s',cache);
-          config = edgeconfig.load({source:cache})
+  edgeconfig.get({ source: source, keys: keys }, function (err, config) {
+    if (err) {
+      const exists = fs.existsSync(cache);
+      console.error("failed to retieve config from gateway. continuing, will try cached copy..");
+      console.error(err);
+      if(!exists){
+        console.error('cache configuration '+cache+' does not exist. exiting.');
+        return;
+      }else{
+        console.log('using cached configuration from %s',cache);
+        config = edgeconfig.load({source:cache})
           if(options.port){
             config.edgemicro.port = parseInt(options.port);
           }
-        }
-      } else {
-        edgeconfig.save(config, cache);
       }
+    } else {
+      edgeconfig.save(config, cache);
+    }
 
-      if (options.cluster) {
-
-        const numWorkers = Number(options.processes || require('os').cpus().length);
-        cluster.setupMaster();
-        const argv = cluster.settings ? cluster.settings.execArgv || [] : [];
-        var j = 0;
-        argv && argv.forEach((arg) => {
-          if (arg.includes('--debug-brk=')) {
-            argv[j] = arg.replace('--debug-brk', '--debug')
-          }
-          j++;
-        })
-        cluster.isMaster && console.log("starting in cluster mode: number workers: " + numWorkers)
-        // Fork workers.
-        for (var i = 0; i < numWorkers; i++) {
-          cluster.fork();
-        }
+    const numWorkers = Number(options.processes || require('os').cpus().length);
+    gateway(config);
+    const naught_options = {
+      'worker-count': numWorkers,
+      'ipc-file': ipcPath,
+      'pid-file': pidPath,
+      'log': clusterStatusLogPath,
+      'stdout': consoleLogPath,
+      'stderr': errorLogPath,
+      'max-log-size': '10485760',
+      'cwd': process.cwd(),
+      'daemon-mode': 'true',
+      'remove-old-ipc': 'true',
+      'node-args': ''
+    };
+    naught.start(naught_options, 'start.js', JSON.stringify(args));
```

Comment:
Sorry about the review open / review close stuff.  Clearly I don't know what I'm doing with the new feature.

```
Error: Cannot find module '/Users/kevin/start.js'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.runMain (module.js:575:10)
    at run (bootstrap_node.js:352:7)
    at startup (bootstrap_node.js:144:9)
    at bootstrap_node.js:467:3
module.js:442
    throw err;
```